components/org.wso2.carbon.secvault/src/main/java/org/wso2/carbon/secvault/internal/SecureVaultActivator.java

```diff
@@ -39,14 +39,21 @@
 
     @Override
     public void start(BundleContext bundleContext) throws Exception {
-        SecureVaultDataHolder.getInstance().setBundleContext(bundleContext);
-        logger.debug("Starting Secure Vault bundle");
-        logger.debug("Initializing Secure Vault config...");
-        Path secureVaultYAMLPath = Utils.getRuntimeConfigPath().resolve(Constants.DEPLOYMENT_CONFIG_YAML);
-        SecureVaultDataHolder.getInstance().setSecureVaultConfiguration(SecureVaultUtils.getSecureVaultConfig
-                (secureVaultYAMLPath).orElseThrow(() -> new SecureVaultException("Error occurred when obtaining " +
-                "secure vault configuration.")));
-        logger.debug("Secure vault config successfully initialized");
+        try {
+            SecureVaultDataHolder.getInstance().setBundleContext(bundleContext);
+            logger.debug("Starting Secure Vault bundle");
+            logger.debug("Initializing Secure Vault config...");
+            Path secureVaultYAMLPath = Utils.getRuntimeConfigPath().resolve(Constants.DEPLOYMENT_CONFIG_YAML);
+            SecureVaultDataHolder.getInstance().setSecureVaultConfiguration(SecureVaultUtils.getSecureVaultConfig
+                    (secureVaultYAMLPath).orElseThrow(() -> new SecureVaultException("Error occurred when obtaining " +
+                    "secure vault configuration.")));
+            logger.debug("Secure vault config successfully initialized");
+        } catch (Throwable throwable) {
+            logger.error("Error occurred when initializing secure vault " + throwable.getMessage(),
```

Comment:
Appending `throwable.getMessage()` doesn't add any value. Exception message will be automatically printed by the logger when printing the `throwable`. So let's remove that. 

 ```java
logger.error("Cannot initialize secure vault.", throwable);
```